components/eamxx/src/share/field/field_utils.hpp

```diff
@@ -139,8 +139,8 @@ void horiz_contraction(const Field &f_out, const Field &f_in,
                    "Error! The weight field must have a column dimension.\n"
                    "The input field has layout "
                        << l_w.tags() << ".\n");
-  EKAT_REQUIRE_MSG(l_in.rank() <= 3,
-                   "Error! The input field must be at most rank-3.\n"
+  EKAT_REQUIRE_MSG(l_in.rank() <= 4,
+                   "Error! The input field must be at most rank-4.\n"
```

Comment:
I think the contractions/reductions should be part of another PR; otherwise, this PR may become huge and hard to review.